core/src/main/scala/org/apache/spark/metrics/source/AccumulatorSource.scala

```diff
@@ -64,10 +64,11 @@ class DoubleAccumulatorSource extends AccumulatorSource
  */
 @Experimental
 object LongAccumulatorSource {
-  def register(sc: SparkContext, accumulators: Map[String, LongAccumulator]): Unit = {
+  def register(sc: SparkContext, accumulators: Map[String, LongAccumulator]): Source = {
```

Comment:
What is the reason of changing this public API and exposing the package private `Source`?

Reply:
it give a chance to remove the registered source of AccumulatorSource's subclass in spark

Reply:
`AccumulatorSource` is package private as well. So to extend it, it will be necessary to place your code within the same package. So there is no need to expose package private `Source` and make it a part of the public contract.

To implement tests without changing the API you could use something like the following
```scala
val source = new LongAccumulatorSource
source.register(accumulators)
sc.env.metricsSystem.registerSource(source)
```
... instead of 
```scala
LongAccumulatorSource.register(mockContext, Map("laF" -> laFirst, "laS" -> laSecond))
```
So the necessary source to delete during the tests will be already stored in the local variable.

Reply:
It makes sense. thanks for your advise.